src/PowerShellEditorServices/Session/PSReadLinePromptContext.cs

```diff
@@ -0,0 +1,216 @@
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using System;
+using System.Management.Automation.Runspaces;
+using Microsoft.PowerShell.EditorServices.Console;
+
+namespace Microsoft.PowerShell.EditorServices.Session {
+    using System.Management.Automation;
+
+    internal class PSReadLinePromptContext : IPromptContext {
+        private const string ReadLineScript = @"
+            [System.Diagnostics.DebuggerHidden()]
+            [System.Diagnostics.DebuggerStepThrough()]
+            param()
+            return [Microsoft.PowerShell.PSConsoleReadLine]::ReadLine(
+                $Host.Runspace,
+                $ExecutionContext,
+                $args[0])";
+
+        // private const string ReadLineScript = @"
```

Comment:
remove or re-consider commented out code

Reply:
Nice catch! Must have missed that one on my sweep :)

Reply:
No worries, as you said, the full review will only come after the conflict resolution and especial when it is not WIP any more.

Reply:
It does indeed :)

Mostly anyway.  The commands we invoke in the background still show up in `Get-History` and still increment `$MyInvocation.HistoryId` which is annoying.  But navigating history with all the PSRL hotkeys like CTRL + R/up/down/etc work as expected in the debugger as well :)